### DISCUSSION TITLE
Fanuc RPC, Controller Requirements

### DISCUSSION THREAD

**Discussion by rastreus** [gavanderhoorn/comet_rpc#3] [Q&A] [answered]
Moved at @gavanderhoorn's request (https://github.com/gavanderhoorn/dominh/discussions/20#discussioncomment-4769676).

---

Is this the Fanuc RPC protocol (`tpc://:3002`) or something different / newer? (https://github.com/onerobotics/go-fanuc/issues/1#issuecomment-502466816)

The project's README mentions:

> FANUC's web server on R-30iB and R-30iB+ controllers (ie: V8.x and up)

_I know it hasn't been tested yet_.

> Requirements are a little unclear at this moment, but it appears `COMET` is installed on all controllers with the base Web Server (HTTP) option and at least V8 of the system software ...

> It's possible R-30iB with V8.x supports the COMET RPC interface as well, but this has not been tested.

Could this _possibly_ work with a PaintMate 200iA 5L with a R-30iB controller (V8.30)? I'm currently using a solution similar to [dominh](https://github.com/gavanderhoorn/dominh) (KCL and `.stm` pages).

**Comment by gavanderhoorn** [comment, marked as answer]
> Is this the Fanuc RPC protocol (`tpc://:3002`) or something different / newer? ([onerobotics/go-fanuc#1 (comment)](https://github.com/onerobotics/go-fanuc/issues/1#issuecomment-502466816))

No, this is not the RPC protocol referred to in the comment/thread you link.

It (ie: `COMET`) offers similar functionality (as in: conceptually, it isn't just another wrapper around the same protocol), though probably lower performance, over a different transport and with a different 'mode' of operation (the PCDK uses the protocol you mention, and that (can) use(s) a sort of scatter-gather approach).

Could you clarify why you are so interested in that protocol specifically?

> The project's README mentions:
> 
> > FANUC's web server on R-30iB and R-30iB+ controllers (ie: V8.x and up)
> 
> _I know it hasn't been tested_, but could this possibly work with a PaintMate 200iA 5L with a R-30iB controller (V8.30)? I'm currently using a solution similar to [dominh](https://github.com/gavanderhoorn/dominh) (KCL and `.stm` pages).

I haven't tested, but `V8.30` should have the `COMET` extension.

I would suggest to "just try it" and see whether it works, and what works.

As the readme already states, `comet_rpc` does not perform any version checks. If something doesn't respond as it's expected to, you'll just get a traceback.

**Reply by gavanderhoorn** [reply]
A comment reporting on the result(s) of any tests you perform would be very much appreciated.

**Reply by gavanderhoorn** [reply]
A quick test with a controller running `V8.30` shows many things work, but some seem to require additional / other arguments. Not sure why or what yet.
